WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,10 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def matching_event_ids_count(event_ids)
+    (self.eventSpecs.split & event_ids).count
```

Comment:
Now take a look at the `def events`, and https://github.com/cubing/worldcubeassociation.org/issues/95 =(